diesel_compile_tests/tests/fail/having_cant_be_used_without_group_by.rs

```diff
@@ -0,0 +1,31 @@
+extern crate diesel;
+
+use diesel::*;
+
+table! {
+    users {
+        id -> Integer,
+        name -> Text,
+    }
+}
+
+table! {
+    posts {
+        id -> Integer,
+        title -> Text,
+        user_id -> Integer,
+    }
+}
+
+joinable!(posts -> users(user_id));
+allow_tables_to_appear_in_same_query!(users, posts);
+
+fn main() {
+    let conn = PgConnection::establish("").unwrap();
+
+    users::table.select(users::name).having(users::id.gt(1)).load(&conn);
+
+    users::table.into_boxed().having(users::id.gt(1)).load(&conn);
+
+    users::table.select(users::name).group_by(users::id).having(posts::id.eq(42)).load(&conn);
```

Comment:
Can we have this test also for boxed select statements?

```suggestion
    users::table.select(users::name).group_by(users::id).having(posts::id.eq(42)).load(&conn);

    users::table.select(users::name).group_by(users::id).into_boxed().having(posts::id.eq(42)).load(&conn);
```

Reply:
Added, thanks to all of you for your patient review and help!